src/bokeh/core/enums.py

```diff
@@ -476,9 +480,6 @@ def enumeration(*values: Any, case_sensitive: bool = True, quote: bool = False)
 #: Specify how textures used as canvas patterns should repeat
 TextureRepetition = enumeration("repeat", "repeat_x", "repeat_y", "no_repeat")
 
-#: Specify how axis tick labels are oriented with respect to the axis
-TickLabelOrientation = enumeration("horizontal", "vertical", "parallel", "normal")
```

Comment:
Breaking change needs a line in the migration guide wiki, I'm not sure if it is possible to make such changes and then drop a link here in the PR to them somehow, but that would be ideal from a process POV. 

Also the feature itself seems like it could be worth explicit mention in the release notes. Just adding a  bullet with a link to the issue or PR would be sufficient, it's much easier to flesh out the release notes if these links are added as we go.

Reply:
I added a release note and updated the wiki.